### PR TITLE
fix: missing global condition

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

1.fix bug of missing global condition
2.compute retention by week/month

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 1]
```diff
@@ -605,41 +606,51 @@ export function getFunnelTableVisualDef(visualId: string, viewName: string, even
...
-    const fieldIdRate = uuidv4();
     if (index === 0) {
-      groupBy.push({
-        NumericalDimensionField: {
-          FieldId: fieldIdRate,
-          Column: {
-            DataSetIdentifier: viewName,
-            ColumnName: 'rate',
-          },
+      continue;
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:610]: why excluding the first event here?

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 2]
```diff
@@ -605,41 +606,51 @@ export function getFunnelTableVisualDef(visualId: string, viewName: string, even
...
+    }
+
+    const fieldIdRate = uuidv4();
+    groupBy.push({
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:614]: which test case(s) is validating those changes?

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:610]: first event don't have conversation rate, we just need a 'event_name' column.

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:614]: this change is let 'rate' as the last column of the chart and rename to 'total_conversation_rate'.  we need check the generated json template if we wan't to check this change.

**Review by tyyzqmf — APPROVED** [review]

**Review by zxkane — COMMENTED** [review]
resolve conflicts locally

**Review by zxkane — APPROVED** [review]

**Pull request merged**